setup.py

```diff
@@ -97,7 +97,7 @@ def build_extension(self, ext):
         "jaxlib>0.4.1",
         "networkx",
         "numpy",
-        "pymbar>=3.0.6,<4",
+        "pymbar>4.0.2",
```

Comment:
nit: worth adding comment explaining why `>4.0.2`?

Reply:
It's to support python 3.12 (https://github.com/choderalab/pymbar/releases/tag/4.0.3)